### DISCUSSION TITLE
G7 - Mismatch wait

### DISCUSSION THREAD

**Discussion by jtrebicki** [NightscoutFoundation/xDrip#3166] [Q&A] [answered]
Hello community!

I have a Dexcom G7 system since a few months; I have a receiver and I use the xDrip+ app.
I have a OnePlus Nord N10 5G with Android 11

I had to place a new sensor today; I then connected it with the receiver, which worked. Then, while the warmup was running, I went into xDrip and stopped the sensor, started a new one and put in the new code. I have done it several times like this, but now I get "Mismatch". (normally I get a notification that I can connect bluetooth, I dont get that now)

 From the docs, this should happen when you put in the wrong code, but it its the correct one or if there is another sensor nearby (there isnt, I disposed it and its far away).

The receiver works flawlessly and I receive the values on it right now

In the main menu I have "Sensor Failed 7"
![image](https://github.com/NightscoutFoundation/xDrip/assets/54892353/e41a5f95-c633-4f6f-a74f-feeb43542e95)

The support page says to put a screenshot of this as well; as said I have done it like that some times, I had it working before
![image](https://github.com/NightscoutFoundation/xDrip/assets/54892353/bdabd9e8-63bf-4bcb-8940-5590f0ed38b4)

This is the status page:
![image](https://github.com/NightscoutFoundation/xDrip/assets/54892353/904e5157-a40d-488a-829c-8282bea8aa82)

I have already tried restarting my phone. And, I have absolutely not mistyped the code, you can see it here on my working receiver
![image](https://github.com/NightscoutFoundation/xDrip/assets/54892353/7e5516c1-8c0c-4ab0-9937-fe98383729e3)

For future reference, if I did something wrong, what is the correct way of replacing the sensor? Should I do xDrip first? 

Thanks in advance for any help!

**Comment by Navid200** [comment]
Forgive me, I read until you said you stopped the sensor and I cannot continue to read because it makes no sense to me.  Why would you stop a G7?  What's the point?  What would go wrong if you leave it alone?
You were going to enter a different transmitter ID, at which point, xDrip would completely forget about the previous device still running.  Why would that be unacceptable?

**Reply by Navid200** [reply]
I have not got my G7s yet.  I will in a few days.  
But, I just performed a test with G6.

I have a G6 running.  I have another G6 transmitter as well, which is currently not running.  I changed the transmitter ID from my active running G6 transmitter to that of my other G6 transmitter, which does not have an active session on it.

As soon as I changed the transmitter ID, checking under the main menu, I can see the start sensor option.  This tells me that xDrip offers you to start if the transmitter ID corresponds to a device that does not have an active session running even if the previous device it was connected to was already running.

I then changed the transmitter ID back to that of my active transmitter.  It connected but was not showing new readings.  I could now see a start sensor option in the menu.  I used it to start sensor.  After the next read cycle, it resumed providing readings.

What this tells me is that we can improve the UI for xDrip.  But, I still see no reason to stop sensor.  I didn't and I just switched devices twice.

**Reply by Navid200** [reply]
I don't believe you need to start xDrip before the receiver.
I'm not sure why you have backed out the xDrip release version in your status page.  
If you are using standard xDrip, you should have no problem showing us which version you are using.

If you are not using the official xDrip release, you should ask the person who has created that version for help.

**Reply by Navid200** [reply]
If you are using the official xDrip, can you disable DND and raise the volume?
Do you not get a pair request?

**Reply by jtrebicki** [reply]
Hello! Thank you for your answer. I stopped the G7 because I just thought I should do that, because in the receiver, I stop the sensor too to start a new session. I didnt know I could just leave it.

I blacked out the version because it said (A screenshot of the system status page (you can blackout serial numbers"
I thought that was sensitive information, my bad. I think I have the newest version, check for updates doesnt show anything; it says 23ccbd1-2023-07-16 where I blacked it out

I had the volume off because I tried to sleep, I had it running multiple hours before without getting a bluetooth request. 
Should I try reinstalling xdrip+? Can I provide any further information to you?
![image](https://github.com/NightscoutFoundation/xDrip/assets/54892353/aac3f5ea-a612-4bad-9a95-77c29dc04e21)
![image](https://github.com/NightscoutFoundation/xDrip/assets/54892353/5fd39ff1-2920-4444-b47b-729293ca69be)
I do not get the request

**Comment by Der-Schubi** [comment]
> Forgive me, I read until you said you stopped the sensor and I cannot continue to read because it makes no sense to me. Why would you stop a G7? What's the point? What would go wrong if you leave it alone? You were going to enter a different transmitter ID, at which point, xDrip would completely forget about the previous device still running. Why would that be unacceptable?

If I remember correctly, xDrip doesn't reset the session if you just change the code to the new sensor in the settings. This leads to false information on first status page, as "Sensor Start" won't be reset.
So you have to stop the old sensor in order to be able to start a new one.

**Reply by Navid200** [reply]
Yes, it does.  I just tested it with G6.  I just changed the transmitter ID and start sensor became available even though the previous device I was connected to still had a sensor running and I never stopped it.

**Reply by Navid200** [reply]
But, the other way around, you are correct.  When I just changed the transmitter id from a transmitter that was stopped to one that was already running, start sensor was still shown and I had to start for xDrip to start showing readings.

I just don't like to stop a sensor if I don't have to.  And I confirmed I didn't have to.  Although, I admit I did this with G6.  
I will have G7 tomorrow and will be able to provide better support.

**Reply by Der-Schubi** [reply]
G7 support is far from perfect, but it works well, if you know the quirks.
As G7 does technically not have to be started or stopped, would there a way of automatically just resetting the sensor start time when changing the transmitter code?

**Reply by Navid200** [reply]
I suggested to remove both start and stop for G7.  It was turned down.  I then discovered that xDrip indeed needs a start just to trigger showing readings.  I am not sure why that is.  Some day hopefully, we can sort that out.

But, stop is what I wish I could remove and after I start using G7, I will see what I can do about it.
It is very difficult to remove anything from xDrip.  It is like pulling teeth!!  The concern always is what if someone is using it?  That's why I always question every new PR.  because I know after it is added, it will be nightmare if I ever want to remove it.

**Comment by Der-Schubi** [comment]
@jtrebicki Mismatch in your case just means, that xDrip found one of your old, discarded or laying around sensors.
You can normally ignore that. But if you are not able to connect to the new sensor, try going out of range of all old / other G7 sensors that may still send.

**Comment by jtrebicki** [comment, marked as answer]
Thanks for every feedback I have gotten!

I resolved the issue, by doing the following:
- Backupped all data from xDrip+
- Uninstalled xDrip+
- Installed it again
- It prompted me to load old settings it found, which I denied
- I activated location all the time again and disabled battery optimization in Android
- Followed G7 setup steps again (https://navid200.github.io/xDrip/docs/Dexcom/G7.html) (didnt change any other settings)

After 5 minutes I got the connection request and I successfully connected again, receiving values now :+1:
